### DISCUSSION TITLE
Why does ImageSharp save the bmp file much larger than the file size saved by Bitmap?

### DISCUSSION THREAD

**Discussion by pedoc** [SixLabors/ImageSharp#2398] [Q&A]
The following is the test code:

```csharp
using System.Buffers;
using System.Diagnostics;
using System.Drawing;
using System.Drawing.Imaging;
using System.Runtime.CompilerServices;
using System.Runtime.InteropServices;
using SixLabors.ImageSharp;
using SixLabors.ImageSharp.PixelFormats;

string file = $@"I:\Image\50befcca14f25-1.bmp";//The image path here can be replaced with any bmp format image
var image = SixLabors.ImageSharp.Image.Load<Rgb24>(file);
image.SaveAsBmp(Path.GetFileNameWithoutExtension(file) + "-imagesharp.bmp");
var bitmap = ImageToBitmap(image);
bitmap.Save(Path.GetFileNameWithoutExtension(file) + "-bitmap.bmp");


static Bitmap ImageToBitmap(SixLabors.ImageSharp.Image<Rgb24> image)
{
    var bitmap = new Bitmap(image.Width, image.Height, PixelFormat.Format24bppRgb);

    for (int y = 0; y < image.Height; y++)
    {
        for (int x = 0; x < image.Width; x++)
        {
            var pixel = image[x, y];
            var color = System.Drawing.Color.FromArgb(pixel.R, pixel.G, pixel.B);
            bitmap.SetPixel(x, y, color);
        }
    }

    // Return the new Bitmap object
    return bitmap;
}
```

Here is the result after saving it as a file. You can see that the size of the file saved by ImageSharp is about 4 times that of the other one. Is this normal?

![image](https://user-images.githubusercontent.com/6604230/224586891-852dc2cf-4099-440f-9a5a-78dcbaab433b.png)

EDIT：

By using the binary tool to compare the generated files, I found that the default format of Bitmap.Save is PNG, and when it is changed to Bmp format, the file size is the same. This explains why the above code has different file sizes. 

Hope it's all helpful for others 😊

**Comment by JimBobSquarePants** [comment]
Can’t help you if you don’t provide the image

**Reply by pedoc** [reply]
thanks

[50befcca14f25-1.zip](https://github.com/SixLabors/ImageSharp/files/10956475/50befcca14f25-1.zip)

**Reply by JimBobSquarePants** [reply]
Sorry, didn't get to investigate before you figured it out. We check the extension against a set of extensions known to the codec so don't make that mistake!
